homeassistant/components/slide_local/cover.py

```diff
@@ -0,0 +1,307 @@
+"""Support for Slide covers."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from goslideapi.goslideapi import (
+    ClientConnectionError,
+    ClientTimeoutError,
+    GoSlideLocal as SlideLocalApi,
+)
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    PLATFORM_SCHEMA as COVER_PLATFORM_SCHEMA,
+    CoverDeviceClass,
+    CoverEntity,
+)
+from homeassistant.const import (
+    ATTR_ID,
+    CONF_API_VERSION,
+    CONF_HOST,
+    CONF_MAC,
+    CONF_PASSWORD,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_TOUCHGO,
+    CONF_INVERT_POSITION,
+    DEFAULT_OFFSET,
+    SERVICE_CALIBRATE,
+    SERVICE_TOUCHGO,
+)
+from .models import SlideConfigEntry
+
+COVER_PLATFORM_SCHEMA = COVER_PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_INVERT_POSITION, default=False): cv.boolean,
+        vol.Optional(CONF_API_VERSION, default=2): cv.byte,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SlideConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up cover(s) for Slide platform."""
+
+    _LOGGER.debug("Initializing Slide cover(s)")
+
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_CALIBRATE,
+        None,
+        "async_calibrate",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_TOUCHGO,
+        {vol.Required(ATTR_TOUCHGO): cv.boolean},
+        "async_touchgo",
+    )
+
+    _LOGGER.debug(
+        "Trying to setup Slide '%s', config=%s",
+        entry.data[CONF_HOST],
+        str(entry),
+    )
+
+    await entry.runtime_data.api.slide_add(
+        entry.data[CONF_HOST],
+        entry.data[CONF_PASSWORD],
+        entry.data[CONF_API_VERSION],
+    )
+
+    try:
+        slide_info = await entry.runtime_data.api.slide_info(entry.runtime_data.host)
+    except (ClientConnectionError, ClientTimeoutError) as err:
+        # https://developers.home-assistant.io/docs/integration_setup_failures/
+
+        _LOGGER.error(
+            "Unable to get information from Slide '%s': %s",
+            entry.runtime_data.host,
+            str(err),
+        )
```

Comment:
We should do this in `__init__.py` and raise `ConfigEntryNotReady` if we can't connect

Reply:
I added it to '__init__.py'. I left it here as well, as slide_info is required to setup the entity.